### ISSUE TITLE
A Question that can be extended to a  feature request.

### ISSUE THREAD

**Issue by Anonymous2716** [xfarrow/locatemydevice#11]
Will this app ever support end to end encryption? Instead of plain text that travels through carriers network.

**Comment by xfarrow** [comment]
That's a nice suggestion that I'll surely take into consideration.
We should implement a public-private keys mechanism

**Comment by kevino-techie** [comment]
well i understand the risk at hand , but how do we even plan to implement that , any ideas ?
since in time of emergency you will be using your contacts phone to send commands but how do you make it e2ee over normal sms. 
Even google rcs requires internet to send e2ee between contacts. Also i don't know if an app can be programmed to view messages from an encrypted chat app.
And i don't think there are ways to implement e2ee on normal sms (also maybe coz of character limits).
I think a password gives most of the security here , since only you know about it and as earlier discussed  it is wise to change password after you had to use it in real scenario.

**Comment by Anonymous2716** [comment]
That's why I titled it a question. I was thinking that over the carriers network data travels unencrypted. So it's only possible if both ends has same app installed right?

**Comment by kevino-techie** [comment]
yeah either locate my device would have to be installed on both device or have to integrate with other apps.
 But my point is , even if you have apps installed on both devices , you would still require internet to exchange those encrypted messages. as it won't happen on normal sms coz of character limit issues.
So its kinda contrary idea to what the app was made for i.e finding phone without any need for secondary app login or internet.
Obviously this feature could be made **optional** if in future we find a solution for implementing it.

**Comment by Anonymous2716** [comment]
What is the limit? And does it vary by countries/carrier?

**Comment by kevino-techie** [comment]
I think the standard limit is 160. ( i referred https://help.clicksend.com/article/h474eseq3a-how-many-characters-can-i-send-in-an-sms )
As per my own findings , the receiver can actually receive the messages as one long text (even though it would be considered as 2 or 3 messages by the carrier ) 
But in this case too, the other user would need to install an app and have to exchange the keys **in advance** for it to work right.

**Comment by kevino-techie** [comment]
okay so so another way of implementing e2ee (as used by the findmydevice app by Nulide) is to read notifications of encrypted chat apps like whatsapp or telegram.
i don't know how well it works but could be added optionally.
